scene/gui/text_edit.cpp

```diff
@@ -1469,15 +1471,15 @@ void TextEdit::_notification(int p_what) {
 										gutter_rect.position.x = size.width - gutter_rect.position.x - gutter_rect.size.x;
 									}
 
-									icon->draw_rect(text_ci, gutter_rect, false, get_line_gutter_item_color(line, g));
+									TS->draw_list_add_texture(draw_list, DRAW_STEP_TEXT, Transform2D(), icon->get_rid(), gutter_rect, get_line_gutter_item_color(line, g));
 								} break;
 								case GUTTER_TYPE_CUSTOM: {
 									if (gutter.custom_draw_callback.is_valid()) {
 										Rect2i gutter_rect = Rect2i(Point2i(gutter_offset, ofs_y), Size2i(gutter.width, row_height));
 										if (rtl) {
 											gutter_rect.position.x = size.width - gutter_rect.position.x - gutter_rect.size.x;
 										}
-										gutter.custom_draw_callback.call(line, g, Rect2(gutter_rect));
+										TS->draw_list_add_custom(draw_list, DRAW_STEP_TEXT, Transform2D(), gutter.custom_draw_callback.bind(line, g, Rect2(gutter_rect)));
```

Comment:
This breaks compatibility since it is exposed, see `set_gutter_custom_draw`. The callback now has 4 params instead of 3.